### ISSUE TITLE
Incorrect (?) use of InstallTrueMethod

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/Semigroups#519]
Semigroups sets the following implication. which seem wrong to me:
```gap
InstallTrueMethod(IsCommutativeSemigroup, IsCommutative);
```
That implies that every commutative magma is a semigroup, which I think is false.

Perhaps this was meant?
```gap
InstallTrueMethod(IsCommutativeSemigroup, IsCommutative and IsSemigroup);
```

**Comment by wilfwilson** [comment]
Hi @fingolfin, thanks for this report. It is true that the meaning that we intend is the second one that you list: `IsCommutativeSemigroup` should be `true` for something that is a semigroup and satisfies `IsCommutative`.

This is technically currently how it works, since `IsCommutativeSemigroup` is declared in `lib/semigrp.gd` as:
```
DeclareProperty("IsCommutativeSemigroup", IsSemigroup);
```
Therefore the `TrueMethod` only applies to things that satisfy `IsSemigroup`. However, if this declaration is ever generalised, which it could conceivably be, then this would lead to problems. So I'll add the `IsSemigroup` to be safe 🙂

**Comment by fingolfin** [comment]
Unfortunately that analysis is incorrect, implications do not (and can not, on a technical level) take into account whether a property was only declared for a subset of all filters. Indeed, create a commutative magma that is not a semigroup, and you'll see this implication was applied to it

**Comment by wilfwilson** [comment]
Ah right, thanks for the correction. Maybe this will discouraging me from making assumptions (and encourage me to verify my statements)!

**Comment by wilfwilson** [comment]
Resolved by #520.
